### PR TITLE
Added documentation for the json key new feature

### PR DESCRIPTION
**Pull request by jselvi**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs

This is part of a pull request modifying several repositories:
- https://github.com/returntocorp/semgrep/pull/6797
- https://github.com/returntocorp/semgrep-langs/pull/24
- https://github.com/returntocorp/semgrep-interfaces/pull/46
- https://github.com/returntocorp/semgrep-docs/pull/843

### CODE REVIEW

File: docs/writing-rules/experiments/extract-mode.md [hunk 1]
```diff
@@ -104,6 +105,10 @@ The `extract` key is required in extract mode. The value must be a metavariable
 
 The `dest-language` key is required in extract mode. The value must be a [language tag](/writing-rules/rule-syntax/#language-extensions-and-tags).
 
+### `json`
+
+The `json` key is optional in extract mode. This boolean value defaults to "false" and it specifies if the extracted content should be parsed as raw source code or as a JSON array. Setting this key to "true" is useful extracting code from json formats such as Jupyter Notebooks.
```
Comment (adamkvitek) [inline docs/writing-rules/experiments/extract-mode.md:110]:
```suggestion
The `json` is an optional key in the extract mode. This boolean value defaults to `false` and specifies whether the extracted content is parsed as raw source code or as a JSON array. Setting the `json` key to `true` is useful in extracting code from JSON formats such as Jupyter Notebooks.
```

File: docs/writing-rules/experiments/extract-mode.md [hunk 2]
```diff
@@ -119,7 +124,7 @@ The value of `reduce` key must be one of the following:
 
 ## Limitations of extract mode
 
-Currently, extract mode does not support additional processing for the extracted text, such as unescaping strings, which might be useful for some applications like Jupyter Notebooks.
+Although extract mode supports JSON array decoding via the `json` key, it does not support other additional processing for the extracted text, such as unescaping strings, which might be useful for some other applications.
```
Comment (adamkvitek) [inline docs/writing-rules/experiments/extract-mode.md:127]:
```suggestion
Although extract mode supports JSON array decoding with the `json` key, it does not support other additional processing for the extracted text, such as unescaping strings.
```

**Review by adamkvitek — COMMENTED** [review]
Thanks, leaving suggestions, please commit them or let me know with further comments. @jselvi

**Comment by jselvi** [comment]
I have committed the changes as suggested. Thanks @adamkvitek for your QA.

**Comment by adamkvitek** [comment]
@jselvi this feature is not out yet?

**Comment by jselvi** [comment]
@adamkvitek The new feature is in another pull request which is linked from this one. It has not been merged yet, no.

**Review by adamkvitek — APPROVED** [review]
Approving so you can merge this PR once related PRs are merged.

**Review by emjin — APPROVED** [review]
lgtm

File: docs/writing-rules/experiments/extract-mode.md [hunk 3]
```diff
@@ -91,8 +91,9 @@ Extract mode rules **also require** two additional fields:
   - `extract`
   - `dest-language`
 
-Extract mode has one **optional** field:
+Extract mode has two **optional** field:
   - `reduce`
+  - `json`
```
Comment (aryx) [inline docs/writing-rules/experiments/extract-mode.md:96]: here it's actually transform:
Thinking about it we should have called it  map, so we have a mapreduce inside extract mode

**Pull request merged**